### PR TITLE
Add Expr->String for WindowFunction

### PR DESCRIPTION
**Pull request by yyy1000**
## Which issue does this PR close?

Related #9726 .

## What changes are included in this PR?

1. Support filter field in aggregate function
2. Support WindowFunction Expr->String

### CODE REVIEW

File: datafusion/sql/src/unparser/expr.rs [hunk 1]
```diff
@@ -355,6 +377,40 @@ impl Unparser<'_> {
...
+            .map(|e| {
+                if matches!(e, Expr::Wildcard { qualifier: None }) {
+                    Ok(ast::FunctionArg::Unnamed(ast::FunctionArgExpr::Wildcard))
+                } else {
+                    self.expr_to_sql(e)
+                        .map(|e| ast::FunctionArg::Unnamed(ast::FunctionArgExpr::Expr(e)))
+                }
+            })
+            .collect::<Result<Vec<_>>>()
+    }
```
Comment (yyy1000) [inline datafusion/sql/src/unparser/expr.rs:412]: Extract a common function for convert function args to ast.

**Review by yyy1000 — COMMENTED** [review]
order_by field in aggregate function and window function is not available now. I plan to do it later.

File: datafusion/sql/src/unparser/expr.rs [hunk 2]
```diff
@@ -199,37 +238,20 @@ impl Unparser<'_> {
...
-                        if matches!(e, Expr::Wildcard { qualifier: None }) {
-                            Ok(FunctionArg::Unnamed(ast::FunctionArgExpr::Wildcard))
-                        } else {
-                            self.expr_to_sql(e).map(|e| {
-                                FunctionArg::Unnamed(ast::FunctionArgExpr::Expr(e))
-                            })
-                        }
-                    })
-                    .collect::<Result<Vec<_>>>()?;
+                let func_name = agg.func_def.name();
```
Comment (alamb) [inline datafusion/sql/src/unparser/expr.rs:241]: 👍

File: datafusion/sql/src/unparser/expr.rs [hunk 3]
```diff
@@ -901,6 +959,53 @@ mod tests {
...
+                    func_def: AggregateFunctionDefinition::BuiltIn(
+                        datafusion_expr::AggregateFunction::Count,
+                    ),
+                    args: vec![Expr::Wildcard { qualifier: None }],
+                    distinct: false,
+                    filter: Some(Box::new(lit(true))),
+                    order_by: None,
+                    null_treatment: None,
+                }),
+                "COUNT(*) FILTER (WHERE true)",
```
Comment (alamb) [inline datafusion/sql/src/unparser/expr.rs:973]: nice

**Review by alamb — APPROVED** [review]
Thank you @yyy1000  -- this looks very nice to me. 

cc @devinjdangelo

**Comment by yyy1000** [comment]
Thanks for your review~ @alamb 
I was busy  with school things last several weeks but now I'm available again. 😆

**Comment by alamb** [comment]
> Thanks for your review~ @alamb I was busy with school things last several weeks but now I'm available again. 😆

It is great to have you back!

**Pull request merged**